code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -0,0 +1,242 @@
+	var/list/wanted_items = list(/obj/item/ectoplasm = 100)
+	///Phrase said when NPC finds none of your inhand items in wanted_items.
+	var/itemrejectphrase = "Sorry, I'm not a fan of anything you're showing me. Give me something better and we'll talk."
+	///Phrase said when you cancel selling a thing to the NPC.
+	var/itemsellcancelphrase = "What a shame, tell me if you changed your mind."
+	///Phrase said when you accept selling a thing to the NPC.
+	var/itemsellacceptphrase = "Pleasure doing business with you."
+	///Phrase said when the NPC finds an item in the wanted_items list in your hands.
+	var/interestedphrase = "Hey, you've got an item that interests me, I'd like to buy it, I'll give you some cash for it, deal?"
+	///Phrase said when the NPC sells you an item.
+	var/buyphrase = "Pleasure doing business with you."
+	///Phrase said when you have too little money to buy an item.
+	var/nocashphrase = "Sorry adventurer, I can't give credit! Come back when you're a little mmmmm... richer!"
+
+/mob/living/simple_animal/hostile/retaliate/trader/attack_hand(mob/user)
+	. = FALSE
+	if(SEND_SIGNAL(src, COMSIG_ATOM_ATTACK_HAND, user) & COMPONENT_CANCEL_ATTACK_CHAIN)
+		. = TRUE
+	var/list/npc_options = list()
+	if(products.len)
+		npc_options += list("Buy" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_buy"))
+	if(lore.len)
+		npc_options += list("Talk" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_talk"))
+	if(wanted_items.len)
+		npc_options += list("Sell" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_sell"))
+	if(!(npc_options.len))
+		return FALSE
+	var/npc_result = show_radial_menu(user, src, npc_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	switch(npc_result)
+		if("Buy")
+			return buy_item(user)
+		if("Sell")
+			return try_sell(user)
+		if("Talk")
+			return deep_lore()
+	face_atom(user)
+
+/**
+ * Checks if the user is ok to use the radial
+ *
+ * Checks if the user is not a mob or is incapacitated or not adjacent to the source of the radial, in those cases returns FALSE, otherwise returns TRUE
+ * Arguments:
+ * * user - The mob checking the menu
+ */
+/mob/living/simple_animal/hostile/retaliate/trader/proc/check_menu(mob/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
+/**
+ * Tries to call sell_item on one of the user's held items, if fail gives a chat message
+ *
+ * Gets both items in the user's hands, and then tries to call sell_item on them, if both fail, he gives a chat message
+ * Arguments:
+ * * user - The mob trying to sell something
+ */
+/mob/living/simple_animal/hostile/retaliate/trader/proc/try_sell(mob/user)
+	var/obj/item/activehanditem = user.get_active_held_item()
+	var/obj/item/inactivehanditem = user.get_inactive_held_item()
+	if(!(sell_item(user, activehanditem)||sell_item(user, inactivehanditem)))
```

Comment:
```suggestion
	if(!sell_item(user, activehanditem) && !sell_item(user, inactivehanditem))
```

Reply:
this does not work like you think it does

Reply:
https://blog.penjee.com/what-is-demorgans-law-in-programming-answered-with-pics/
are you sure

Reply:
![image](https://user-images.githubusercontent.com/23585223/114831876-271d7d80-9dce-11eb-82ac-d0075ed2e6f5.png)